basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -69,40 +69,54 @@ namespace basic_autonomy
                                                                 const carma_wm::WorldModelConstPtr &wm, const GeneralTrajConfig &general_config, 
                                                                 const DetailedTrajConfig &detailed_config, std::unordered_set<lanelet::Id> &visited_lanelets)
         {
-            if(maneuver.type != cav_msgs::Maneuver::LANE_FOLLOWING){
+            if (maneuver.type != cav_msgs::Maneuver::LANE_FOLLOWING)
+            {
                 throw std::invalid_argument("Create_lanefollow called on a maneuver type which is not LANE_FOLLOW");
             }
             std::vector<PointSpeedPair> points_and_target_speeds;
 
             cav_msgs::LaneFollowingManeuver lane_following_maneuver = maneuver.lane_following_maneuver;
 
-            auto lanelets = wm->getLaneletsBetween(starting_downtrack, lane_following_maneuver.end_dist + detailed_config.buffer_ending_downtrack, true, true);
-
-            //TODO: This fix is temporary and only applies on lane follow maneuvers from Platooning Strategic Plugin IHP. 
-            // A general fix will be implemented soon. (issue #1863)
             bool lanelets_defined = !maneuver.lane_following_maneuver.lane_ids.empty();
-            ROS_DEBUG_STREAM("lanelets_defined: " << lanelets_defined);
-            bool isFromPlatooning = maneuver.lane_following_maneuver.parameters.planning_strategic_plugin == "PlatooningStrategicIHPPlugin";
-            ROS_DEBUG_STREAM("isFromPlatooning: " << isFromPlatooning);
 
-            if (lanelets_defined && isFromPlatooning)
+            if (!lanelets_defined)
```

Comment:
This could be reduced to 
```
if (maneuver.lane_following_maneuver.lane_ids.empty())
```
To remove the intermediate variable and need for double negation.

Reply:
changed